File: app/src/main/java/com/gmail/fattazzo/formula1livenews/fragments/home/driverstandings/DriverStandingsListAdapter.java
package com.gmail.fattazzo.formula1livenews.fragments.home.driverstandings;

import android.content.Context;
import android.view.View;
import android.view.ViewGroup;
import android.widget.BaseAdapter;

import com.gmail.fattazzo.formula1livenews.ergast.objects.DriverStandings;

import org.androidannotations.annotations.AfterInject;
import org.androidannotations.annotations.EBean;
import org.androidannotations.annotations.RootContext;

import java.util.ArrayList;
import java.util.List;

@EBean
class DriverStandingsListAdapter extends BaseAdapter {

    private List<DriverStandings> drivers;

    @RootContext
    Context context;

    public void setDrivers(List<DriverStandings> drivers) {
        this.drivers = drivers;
    }

    @AfterInject
    void initAdapter() {
        drivers = new ArrayList<>();
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        DriverStandingsItemView driverStandingsItemView;
        if (convertView == null) {
            driverStandingsItemView = DriverStandingsItemView_.build(context);
        } else {
            driverStandingsItemView = (DriverStandingsItemView) convertView;
        }

        driverStandingsItemView.bind(getItem(position));

        return driverStandingsItemView;
    }

    @Override
    public int getCount() {
        return drivers.size();
    }

    @Override
    public DriverStandings getItem(int position) {
        return drivers.get(position);
    }

    @Override
    public long getItemId(int position) {
        return position;
    }

    void clearItems() {
        drivers.clear();
    }
}

File: app/src/main/kotlin/com/gmail/fattazzo/formula1world/fragments/home/driverstandings/DriverStandingsListAdapter.kt
/*
 * Project: total-gp-world
 * File: DriverStandingsListAdapter.kt
 *
 * Created by fattazzo
 * Copyright © 2018 Gianluca Fattarsi. All rights reserved.
 *
 * MIT License
 * Permission is hereby granted, free of charge, to any person obtaining a copy
 * of this software and associated documentation files (the "Software"), to deal
 * in the Software without restriction, including without limitation the rights
 * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
 * copies of the Software, and to permit persons to whom the Software is
 * furnished to do so, subject to the following conditions:
 *
 * The above copyright notice and this permission notice shall be included in all
 * copies or substantial portions of the Software.
 *
 * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
 * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
 * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
 * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
 * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
 * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
 * SOFTWARE.
 */

package com.gmail.fattazzo.formula1world.fragments.home.driverstandings

import android.content.Context
import android.view.View
import android.view.ViewGroup
import android.widget.BaseAdapter
import com.gmail.fattazzo.formula1world.domain.F1DriverStandings
import org.androidannotations.annotations.AfterInject
import org.androidannotations.annotations.EBean
import org.androidannotations.annotations.RootContext
import java.util.*

@EBean
internal open class DriverStandingsListAdapter : BaseAdapter() {

    private var drivers: MutableList<F1DriverStandings>? = null

    @RootContext
    lateinit var context: Context

    fun setDrivers(drivers: MutableList<F1DriverStandings>) {
        this.drivers = drivers
    }

    @AfterInject
    fun initAdapter() {
        drivers = ArrayList()
    }

    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {

        val driverStandingsItemView: DriverStandingsItemView
        if (convertView == null) {
            driverStandingsItemView = DriverStandingsItemView_.build(context)
        } else {
            driverStandingsItemView = convertView as DriverStandingsItemView
        }

        driverStandingsItemView.bind(getItem(position))

        return driverStandingsItemView
    }

    override fun getCount(): Int {
        return drivers!!.size
    }

    override fun getItem(position: Int): F1DriverStandings {
        return drivers!![position]
    }

    override fun getItemId(position: Int): Long {
        return position.toLong()
    }

    fun clearItems() {
        drivers!!.clear()
    }
}